src/components/network-map-viewer/network/draw-control.ts

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import MapboxDraw from '@mapbox/mapbox-gl-draw';
+import { UUID } from 'crypto';
+import { useCallback } from 'react';
+import { useControl } from 'react-map-gl';
+
+import type { ControlPosition } from 'react-map-gl';
+
+let mapDrawerController: MapboxDraw | undefined = undefined;
+
+export function getMapDrawer() {
+    return mapDrawerController;
+}
+
+//source: https://github.com/visgl/react-map-gl/blob/master/examples/draw-polygon/src/
+type DrawControlProps = ConstructorParameters<typeof MapboxDraw>[0] & {
+    position?: ControlPosition;
+    readyToDisplay: boolean;
+    studyUuid: UUID;
+    mapEquipments: any;
+    geoData: any;
+    onDrawModeChanged: (polygoneDrawing: boolean) => void;
+    onUpdate: (e: any) => void;
+    onDelete: (e: any) => void;
+};
+
+export default function DrawControl(props: DrawControlProps) {
+    const onModeChange = useCallback((e: any) => {
+        if (e.mode === 'draw_polygon') {
+            props.onDrawModeChanged(true);
+        } else {
+            // mode === 'simple_select'
+            props.onDrawModeChanged(false);
+        }
+    }, []);
```

Comment:
why not forwarding all the ModeChange event to the parent ?
What does `onDrawModeChanged(true);` means ?
the mode has changed ? seems strange to me